lib/quicly.c

```diff
@@ -873,6 +884,31 @@ void quicly_get_max_data(quicly_conn_t *conn, uint64_t *send_permitted, uint64_t
         *consumed = conn->ingress.max_data.bytes_consumed;
 }
 
+static void update_idle_timeout(quicly_conn_t *conn, int is_in_send)
+{
+    if (is_in_send && !conn->idle_timeout.should_rearm_on_send)
+        return;
+
+    int64_t idle_msec = INT64_MAX;
+    /* TODO reconsider how to refer to peer's idle-timeout value after https://github.com/quicwg/base-drafts/issues/2602 gets
+     * resolved */
+    if (conn->initial == NULL && conn->handshake == NULL && conn->super.peer.transport_params.idle_timeout != 0)
+        idle_msec = conn->super.peer.transport_params.idle_timeout;
+    if (conn->super.ctx->transport_params.idle_timeout != 0 && conn->super.ctx->transport_params.idle_timeout < idle_msec)
+        idle_msec = conn->super.ctx->transport_params.idle_timeout;
+
+    if (idle_msec == INT64_MAX)
+        return;
+
+    conn->idle_timeout.at =
+        now + idle_msec + 3 * quicly_rtt_get_pto(&conn->egress.loss.rtt, conn->super.ctx->transport_params.max_ack_delay);
+    if (is_in_send) {
```

Comment:
It's likely that the branch is optimized out, but what do you think about:
`conn->idle_timeout.should_rearm_on_send = !is_in_send;`

Reply:
Will change.

The downside is that the change makes it harder to set non-conditional breakpoints (conditional breakpoints incur performance overhead), but I think we do not need to care about that here.

Reply:
Addressed in f8fce03.